data_preparation/data_cleaning.py

```diff
@@ -215,9 +207,17 @@ def fill_columns_with_missing_values(df):
             df[column] = df.groupby(['make', 'model', 'trim', 'manufactured_year'])[column].transform(lambda x: x.fillna(x.mode()[0] if not x.mode().empty else np.nan))
             df[column] = df.groupby(['make', 'model', 'trim'])[column].transform(lambda x: x.fillna(x.mode()[0] if not x.mode().empty else np.nan))
     
-    return df    
+    return df
+
+def train_no_of_owners_model(df: pd.DataFrame) -> np.ndarray:
```

Comment:
Why have you added this function?

Reply:
Also whats the difference between me adding this comment and starting a review

Reply:
I removed this change in a later commit. If you go on the Files Changed tab and review each file then it'll show you the final version of changes I made. If you click on a specific commit and review one of those changes (as you've done here) then you're not necessarily looking at the latest changes on that file i.e. I could have reverted that change in a later commit (which is the case here). I think you've done this on one of you other comments, I'll just say `SEE FINAL FILE VERSION` in reply where you've done it.

And what you're doing now (writing comments) is the reviewing propose. You can propose further changes etc and then when you're finally happy with them you can add your approval, which is you saying that you're okay with it being merged into master. Then it's up to me to merge it into master (you call you master branch main for this repo actually). Cause I'm a collaborator on this repo I can merge into master without approval also. But making a PR and reviewing is good practice especially for non-trivial changes.

Reply:
Nice